File: LAB QUIZZES/CSE221_Quiz01/_A1E/_A1E.java
import java.util.*;
import java.io.*;

public class _A1E {
    public static Random rng = new Random();
    public static int test = 0, best = 0, score = 0, batch = 0, total = 0;
    public static int timeLimit = 2, nBatch = 5, N = 0;
    public static int weight[] = { 0, 1, 2, 2, 2, 3 };
    public static int nTest[] = { 0, 1000000, 10000, 100, 500, 50 };
    public static int maxN[] = { 0, 10, 100, 1000, 10000, 100000 };
    public static ArrayList<Integer> arr = new ArrayList<>(), res = new ArrayList<>();

    public static void printCase(String verdict, int test, ArrayList<Integer> arr)
            throws Exception {
        if (new File(verdict + ".txt").length() == 0) {
            BufferedWriter fout = new BufferedWriter(new FileWriter(verdict + ".txt"));
            fout.write("batch = " + batch + "\ntest = " + test + "\n");
            fout.write("N = " + arr.size() + "\n");
            fout.write("arr = " + arr + "\n");
            fout.close();
        }
        System.out.print(verdict + " on Batch " + batch + "\n");
        System.exit(1);
    }

    public static void checkSoln(ArrayList<Integer> res)
            throws Exception {
        ArrayList<Integer> brr = new ArrayList<>(res);
        Collections.sort(brr);
        if (!res.equals(brr))
            printCase("WrongAnswer", test, new ArrayList<>(arr));
    }

    public static void limitTime() {
        try {
            Thread.sleep(1000 * timeLimit);
            printCase("TimeLimitExceeded", test, new ArrayList<>(arr));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static void main(String args[]) throws Exception {
        if (args.length == 1)
            rng = new Random(batch = Integer.parseInt(args[0]));
        if (args.length != 1 || batch < 1 || nBatch < batch) {
            BufferedReader fin = new BufferedReader(new FileReader("EnterIDandLanguage.txt"));
            String ID = fin.readLine().split("\\s+", 2)[0];
            fin.close();
            if (new File("_A1E_" + ID + ".java").exists()) {
                fin = new BufferedReader(new FileReader("_A1E_" + ID + ".java"));
                best = Integer.parseInt(fin.readLine().split("\\s+", 4)[2]);
                fin.close();
            }
            new FileOutputStream("WrongAnswer.txt").close();
            new FileOutputStream("TimeLimitExceeded.txt").close();
            for (batch = 1; batch <= nBatch; total += weight[batch], ++batch) {
                Process process = new ProcessBuilder("java", "_A1E", "" + batch).start();
                fin = new BufferedReader(new InputStreamReader(process.getInputStream()));
                fin.lines().forEach(System.out::println);
                fin.close();
                fin = new BufferedReader(new InputStreamReader(process.getErrorStream()));
                fin.lines().forEach(System.out::println);
                fin.close();
                if (process.waitFor() == 0)
                    score += weight[batch];
            }
            if (best <= score) {
                BufferedWriter fout = new BufferedWriter(new FileWriter("_A1E_" + ID + ".java"));
                fout.write("// " + ID + " " + score + "\n");
                fout.close();
                new ProcessBuilder("cmd", "/c", "echo // %COMPUTERNAME% %USERNAME%>>_A1E_" + ID + ".java").start()
                        .waitFor();
                fout = new BufferedWriter(new FileWriter("_A1E_" + ID + ".java", true));
                fin = new BufferedReader(new FileReader("_A1E_Solution.java"));
                fout.write(fin.lines().collect(java.util.stream.Collectors.joining(System.lineSeparator())));
                fin.close();
                fout.close();
            }
            System.out.print("Tentative score = " + (score / (double) total) + "/1\n");
            System.exit(0);
        }
        System.out.println("Running on Batch " + batch);
        long start = System.nanoTime();
        new Thread(() -> limitTime()).start();
        if (1 <= batch && batch <= nBatch) {
            for (test = 1; test <= nTest[batch]; ++test) {
                res.clear();
                arr.clear();
                N = 1 + rng.nextInt(maxN[batch]);
                arr.ensureCapacity(N);
                for (int i = 1; i <= N; ++i)
                    arr.add(1 + rng.nextInt(1000000000));
                if (1000 < N)
                    arr.sort(null);
                res = _A1E_Solution.solve(new ArrayList<>(arr));
                checkSoln(res);
            }
        }
        long finish = System.nanoTime();
        long elapsed = finish - start;
        System.out.printf("Passed Batch " + batch + " in %.9fs\n", elapsed * 1e-9);
        System.exit(0);
    }
}

File: LAB QUIZZES/CSE221_Quiz01/_A1E/_A1E.py
import os, sys, time, random, threading, subprocess
from _A1E_Solution import solve
test, best, score, batch, total = 0, 0, 0, 0, 0
timeLimit, nBatch, N = 2, 5, 0
weight = [0, 1, 2, 2, 2, 3]
nTest = [0, 1000000, 10000, 100, 500, 50]
maxN = [0, 10, 100, 1000, 10000, 100000]
arr, res = [], []
def printCase(verdict, test, arr):
    if os.path.getsize(verdict + ".txt") == 0:
        with open(verdict + ".txt", "w") as fout:
            fout.write("batch = " + str(batch) + "\ntest = " + str(test) + "\n")
            fout.write("N = " + str(len(arr)) + "\n")
            fout.write("arr = " + str(arr) + "\n")
    print(verdict + " on Batch " + str(batch), flush=True)
    os._exit(1)
def checkSoln(res):
    if not all(res[i] <= res[i + 1] for i in range(len(res) - 1)):
        printCase("WrongAnswer", test, arr.copy())
def limitTime():
    time.sleep(timeLimit)
    printCase("TimeLimitExceeded", test, arr.copy())
if len(sys.argv) == 2:
    batch = int(sys.argv[1])
    random.seed(batch)
if len(sys.argv) != 2 or batch < 1 or nBatch < batch:
    ID = ""
    with open("EnterIDandLanguage.txt") as fin:
        ID = fin.readline().split()[0]
    if os.path.exists("_A1E_" + ID + ".py"):
        with open("_A1E_" + ID + ".py") as fin:
            best = int(fin.readline().split()[2])
    with open("WrongAnswer.txt", "w") as fout:
        pass
    with open("TimeLimitExceeded.txt", "w") as fout:
        pass
    batch = 1
    while batch <= nBatch:
        process = subprocess.run("pypy _A1E.py " + str(batch), capture_output=True, text=True)
        if process.stdout:
            print(process.stdout, end="")
        if process.stderr:
            print(process.stderr, end="")
        if process.returncode == 0:
            score += weight[batch]
        total += weight[batch]
        batch += 1
    if best <= score:
        with open("_A1E_" + ID + ".py", "w") as fout:
            fout.write("# " + ID + " " + str(score) + "\n")
        subprocess.run("cmd /c echo # %COMPUTERNAME% %USERNAME%>>_A1E_" + ID + ".py")
        with open("_A1E_" + ID + ".py", "a") as fout:
            with open("_A1E_Solution.py") as code:
                fout.write(code.read())
    print("Tentative score = " + str(score / total) + "/1", flush=True)
    os._exit(0)
print("Running on Batch " + str(batch), flush=True)
start = time.time()
threading.Thread(target=limitTime, daemon=True).start()
if 1 <= batch and batch <= nBatch:
    test = 1
    while test <= nTest[batch]:
        res = []
        arr = []
        N = random.randint(1, maxN[batch])
        i = 1
        while i <= N:
            arr.append(random.randint(1, 1000000000))
            i += 1
        if 1000 < N:
            arr.sort()
        res = solve(arr.copy())
        checkSoln(res)
        test += 1
finish = time.time()
elapsed = finish - start
print("Passed Batch " + str(batch) + " in " + f"{elapsed:.9f}" + "s", flush=True)
os._exit(0)